exporter/datadogexporter/internal/testutil/logs_agent.go

```diff
@@ -0,0 +1,82 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package testutil // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/testutil"
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+)
+
+type DatadogLogsAgentServer struct {
+	*httptest.Server
+	// LogsData is the array of json requests sent to datadog backend
+	LogsData          JSONLogs
+	connectivityCheck sync.Once
+	doneChannel       chan bool
+}
+
+// DatadogLogsAgentServerMock mocks a Datadog Logs Intake backend server for the logs agent
+func DatadogLogsAgentServerMock(doneChannel chan bool, overwriteHandlerFuncs ...OverwriteHandleFunc) *DatadogLogsAgentServer {
```

Comment:
Why do we need a different mock server? can't we reuse the existing one ?

Reply:
I tried reusing the existing one but when I added the new "/api/v2/logs" endpoint, the existing `TestLogsExporter` tests started sending to "/api/v2/logs" instead of "/", causing the existing tests to fail.